src/main/java/org/jfrog/hudson/ArtifactoryBuilder.java

```diff
@@ -381,16 +382,32 @@ private boolean isEmptyUrls(List<JFrogPlatformInstance> jfrogInstances) {
             return false;
         }
 
-        private boolean fillEmptyServers(List<JFrogPlatformInstance> jfrogInstances) {
+        private void autoFillPlatformServers(List<JFrogPlatformInstance> jfrogInstances) {
             if (jfrogInstances == null) {
-                return false;
+                return;
             }
             for (JFrogPlatformInstance instance : jfrogInstances) {
                 if (StringUtils.isBlank(instance.getArtifactoryServer().getArtifactoryUrl())) {
                     instance.getArtifactoryServer().setArtifactoryUrl(instance.getUrl() + "/artifactory");
+                    continue;
+                }
+                if (StringUtils.isBlank(instance.getUrl())) {
+                    continue;
+                }
+                // Check if Artifactory URL has a different prefix than platform URL.
+                if (!StringUtils.startsWithIgnoreCase(instance.getArtifactoryServer().getArtifactoryUrl(), instance.getUrl())) {
+                    // Try to search JFrog instance by the current Artifactory URL.
+                    Optional<JFrogPlatformInstance> oldInstance = this.jfrogInstances.stream().filter(currentInstance -> currentInstance.getArtifactoryServer().getArtifactoryUrl().equals(instance.getArtifactoryServer().getArtifactoryUrl()) && currentInstance.getId().equals(instance.getId())).findFirst();
```

Comment:
We are in the context of a specific jfrog instance. Why then should we iterate over all configured instances? I don't really understand the code, and the description comment doesn't help understand this.